include/drivers/gpio/nrfe_gpio.h

```diff
@@ -32,6 +33,11 @@ typedef struct __packed {
 			 */
 } nrfe_gpio_data_packet_t;
 
+typedef struct __packed {
+	nrfe_shared_data_lock_t lock;
+	nrfe_gpio_data_packet_t data;
+} nrfe_gpio_mbox_data_t;
+
```

Comment:
It should probably be defined only for MBOX backend. Same goes for `nrfe_shared_data_lock_t`

Reply:
Look here: https://github.com/nrfconnect/sdk-nrf/pull/16592#discussion_r1750365085
It is a typedef, so if it is not used it will have no effect on the code.